src/server/protocol.ts

```diff
@@ -232,6 +232,12 @@ namespace ts.server.protocol {
          * Contains extra information that plugin can include to be passed on
          */
         metadata?: unknown;
+
+        /**
+         * Time spent updating the program graph, in milliseconds.
+         */
+        /* @internal */
+        updateGraphDurationMs?: number;

```

Comment:
Consider making this property an object that has an updateGraphDurationMs in it. If we want to put in more per request diagnostics or metrics it would be easy to extend that. Currently it seems like a very specific property to have on an otherwise generic response object.

Reply:
Introduced `performanceData`.